### ISSUE TITLE
Work party capacity not working correctly: Can't add from waitlist

### ISSUE THREAD

**Issue by WaBirder57** [washingtontrails/vms#169]
[Bug report from Julie Cassata]

There's something wrong with how capacity it working.

Here is a work party with capacity of 4 - 
https://cs9.salesforce.com/a2eK0000000wPoiIAE

1.	I filled the work party until it was full. It worked correctly, switching to waitlist once 4 were registered.
2.	One registered participant, Al Pine, cancelled, logged in, from VMS. 
3.	I received the email notifying me to invite waitlisted folk. (I'm sending an email separately about that pointing out another error.) So far, so good.
4.	From the work party record in SF, I try to change the status of the single person that is on the waitlist, Seymour Mountains, and it does not let me. Isn't that the correct way to add someone from the waitlist? (Changing them from waitlisted to "invited from waitlist"?) Totally not working. I get this error, even though room was just made for another.

"Error:You cannot register due to insufficient work party capacity."
.
One thing that doesn't seem right, which may be contributing to the error is that "number of remaining spots" is 0, when I would think it should say 1, once someone cancels. Perhaps there's something wrong with how it is/isn't counting the crew leader.

I think that's what's happening. The little help bubble for "number of remaining spots," includes, "Saves 1 space for the CL." Since we corrected it to create a registration for the crew leader, I bet it's double counting the CL ! (Slam dunk by Julie!)

**Comment by mscholtz** [comment]
OK, I think I see the problem.  And good guess by Julie, but I don't think the crew leader is the issue here.

We put in two validation steps that I believe are conflicting with each other.

1) It includes Waitlisted people when calculating Spot Left.  We did this so that when someone cancels (like happened here) some new registrant on the site couldn't snag the open spot before staff had a chance to offer it to the waitlist.  So it effectively holds spots for people on the waitlist.  So showing Remaining Spots = 0 is what we want, and expected.

2)  We also recently did ticket #134 which does a check against capacity when switching the status of a registration.  That's the problem here.  We weren't thinking about (1) when we did this.  We do want to do a capacity check, but because of (1), when changing from Waitlisted to Invited, or from Invited to Registered, often Spots Remaining is going to be 0.  

I think the fix here is to have two different numbers for Remaining Spots.  One that includes people on the waitlist and one that doesn't.  Right now we only have the former.  That should still be used for determining if new registrants go on the waitlist.  But the latter should be used to determine if it's OK to switch a person from Waitlisted to Invited, or from Invited to Registered.
